### DISCUSSION TITLE
Accessing state of Pinia store in boot files always gives null

### DISCUSSION THREAD

**Discussion by straurob** [quasarframework/quasar#13648] [General - Components / Directives / etc] [answered]
I'm having some trouble getting Pinia to work properly with two boot different files.

Boot file 1 (`authentication.js`)  fetches information about the current user and replaces the state in the `userInfoStore`:

```javascript
import { boot } from 'quasar/wrappers';
import userClient from 'src/api/userClient';
import { useUserInfoStore } from 'stores/userInfo';

export default boot(async ({ router, store }) => {
  const userInfoStore = useUserInfoStore(store);
  router.beforeEach(async () => {
    try {
      const response = await userClient.getUserInfo();
      userInfoStore.$state = response.data;
      return true;
    } catch (error) {
        // error handling
        return false;
    }
    return true;
  });
});
```

Then, in boot file 2 (`casl.js`) I'd like to read the state from the `userInfoStore`like this:

```javascript
import { boot } from 'quasar/wrappers';
import { useUserInfoStore } from 'stores/userInfo';

export default boot(async ({ store }) => {
  const userInfoStore = useUserInfoStore();
  // this prints null  
  console.log(userInfoStore.givenName);
  // this also gives null  
  console.log(store.state.userInfo);
});
```

The `givenName` is always `null`. Inspecting the store via the Vue DevTools, however, shows a `givenName`. And printing `console.log(userInfoStore.$state)` also shows the name.

The `quasar.config.js` defines this order of boot files:

`boot: ['authentication', 'casl']`

[![enter image description here][1]][1]

What could be wrong here?

  [1]: https://i.stack.imgur.com/PaWcA.png

**Comment by pdanpdan** [comment, marked as answer]
Why would you expect it not to be null?
In the first boot file you are registering a hook on navigation that updates the store when the navigation happens.

In the second boot file you are reading from store, but boot files are executed before navigation.

**Reply by straurob** [reply]
Oh my goodness... yes, of course. I couldn't see the forest for the trees. I will follow your proposal and move both actions in the rotuer hook.

**Comment by pdanpdan** [comment]
Put them both in the same boot file, and do your casl related things in the same router hook.
